lib/network_utils.pm

```diff
@@ -72,7 +72,7 @@ Returns if can ping worker host gateway
 sub can_upload_logs {
     my ($gw) = @_;
     $gw ||= testapi::host_ip();
-    return (script_run('ping -c 1 ' . $gw) == 0);
+    return (script_run('ping -c 3 ' . $gw, 60) == 0);
```

Comment:
I suggest increase timeout for both script_run and ping but keep "-c 1"
script_run('time ping -c 1 -W 120' . $gw , timeout => 120)

Reply:
the problem is that no matter how big the timeout is, it will always sometimes fail. That is because when ping returns, sometimes the return code is not captured by openQA. When that happens, even if the ping returns successfully in 5 seconds, openQA will just wait for 120 seconds to say timed out.

I must say that we probably hit a bug that goes deep in the os-autoinst tooling with this one.

Reply:
ok, let's give a try in openqa.
Will be more confidence if we have more validation result.

Reply:
but wait a second, isn't this the same thing that we have fixed in a couple of places?
that fqdn seems to misbehave, we should use the ip of the worker.

Reply:
> but wait a second, isn't this the same thing that we have fixed in a couple of places? that fqdn seems to misbehave, we should use the ip of the worker.

This issue is unrelated to the other one (where fqdn was not misbehaving but it was that the ibft test needed an ip instead of fqdn, but there was recently a change where WORKER_HOSTNAME was no longer providing ip but fqdn). 

The problem here is that some times `script_run` times out (not the return code being a wrong value). It sometimes times out because it does not catch the return code, even though ping has returned. Ping returns fairly quickly, return code is not caught and script_run times out when the timeout duration is expired.

Reply:
as you described looks like some problem in os-autoinst.. a couple of questions:
(1) could you try with IP if we could find better rate of success for this unknown issue of script_run (if doesn't need to resolve anything, perhaps it is faster and the serial is read faster, just a theory...).
(2) does it help to increase the timeout or not? I don't get from the first comment if there is any improvement.

Reply:
On the second point, increasing timeout does not help, since the ping command returns quite quickly either way.
On the first point, I can try with IP, but I am convinced this is unrelated to speed of the command or how fast the serial output is read, it is just that, when it fails, it is never read, although the command has returned and we are on the next line on the command prompt.

Reply:
thanks, if you can take a look at least we discard that.
So if the timeout increase doesn't help do we need to merge this PR?

Reply:
I tried pinging IP instead of fqdn, still manages to fail some times.